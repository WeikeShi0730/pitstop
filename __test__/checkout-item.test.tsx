import {
  render,
  RenderResult,
  screen,
  fireEvent,
} from "@testing-library/react";
import CheckoutItem from "../components/checkout-item.component";

let documentBody: RenderResult;
describe("<CheckoutItem />", () => {
  it("shows content in <CheckoutItem />", () => {
    documentBody = render(
      <CheckoutItem
        cartItem={{
          product: {
            imageUrl: "test",
            name: "test name",
            price: 0,
            id: "",
            featured: false,
            sold: 0,
            teamId: "",
          },
          count: 10,
        }}
      />
    );
    expect(documentBody.queryByText("test name")).toBeInTheDocument();
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });

  it("shows <CheckoutItem /> add on click", async () => {
    documentBody = render(
      <CheckoutItem
        cartItem={{
          product: {
            imageUrl: "test",
            name: "test name",
            price: 0,
            id: "",
            featured: false,
            sold: 0,
            teamId: "",
          },
          count: 10,
        }}
      />
    );
    // need to sign in firebase first
    expect(await screen.findByDisplayValue("10")).toBeInTheDocument();
    expect(await screen.findByText("test name")).toBeInTheDocument();
    const button = await screen.findByRole("button", { name: /âŠ•/i });
    // fireEvent.click(button);
    // expect(await screen.findByDisplayValue("11")).toBeInTheDocument();
  });
  it("shows <CheckoutItem /> remove on click", async () => {
    documentBody = render(
      <CheckoutItem
        cartItem={{
          product: {
            imageUrl: "test",
            name: "test name",
            price: 0,
            id: "",
            featured: false,
            sold: 0,
            teamId: "",
          },
          count: 10,
        }}
      />
    );
    // need to sign in firebase first
    expect(await screen.findByDisplayValue("10")).toBeInTheDocument();
    expect(await screen.findByText("test name")).toBeInTheDocument();
    const button = await screen.findByRole("button", { name: /âŠ–/i });
    // fireEvent.click(button);
    // expect(await screen.findByDisplayValue("9")).toBeInTheDocument();
  });
  it("shows <CheckoutItem /> delete on click", async () => {
    documentBody = render(
      <CheckoutItem
        cartItem={{
          product: {
            imageUrl: "test",
            name: "test name",
            price: 0,
            id: "",
            featured: false,
            sold: 0,
            teamId: "",
          },
          count: 10,
        }}
      />
    );
    // need to sign in firebase first
    expect(await screen.findByDisplayValue("10")).toBeInTheDocument();
    expect(await screen.findByText("test name")).toBeInTheDocument();
    const button = await screen.findByRole("button", { name: /ðŸ…§/i });
    // fireEvent.click(button);
    // expect(await screen.findByDisplayValue("9")).toBeInTheDocument();
  });
});
