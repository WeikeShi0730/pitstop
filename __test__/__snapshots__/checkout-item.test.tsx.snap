// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutItem /> shows <CheckoutItem /> add on click 1`] = `
<body>
  <div>
    <div
      class="flex justify-center w-full sm:w-2/3 mx-5 text-slate-700 text-sm md:text-base"
      id="checkoutItem"
    >
      <div
        class="flex w-full bg-opacity-30 bg-slate-100 rounded-lg transition-all duration-200 ease-in-out shadow-md hover:shadow-slate-500"
      >
        <div
          class="image flex flex-col relative w-1/3 h-48"
        >
          <div
            class="relative w-full h-full bg-[#F8F8F8] rounded-l-lg"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="test name image"
                class="object-contain rounded-l-lg"
                data-nimg="fill"
                decoding="async"
                src="test"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </span>
          </div>
        </div>
        <div
          class="text flex flex-col justify-evenly mx-4 w-2/3"
        >
          <div
            class="title flex justify-start"
          >
            <div
              class="font-normal text-lg md:text-2xl"
            >
              test name
            </div>
          </div>
          <div
            class="details flex flex-col md:flex-row items-center justify-between"
          >
            <div
              class="unt flex items-end justify-start text-sm md:text-base text-slate-500"
            >
              <div>
                CAD
              </div>
              <div
                class="text-base md:text-xl ml-1"
              >
                0
              </div>
              <div
                class=""
              >
                /ea
              </div>
            </div>
            <div
              class="qty flex items-center"
            >
              <div
                class="text-sm md:text-base"
              >
                Qty:
              </div>
              <div
                class="mx-3 text-base md:text-xl"
              >
                <button
                  id="remove"
                  name="REMOVE"
                >
                  âŠ–
                </button>
              </div>
              <div
                class="flex justify-center text-base md:text-xl"
              >
                <form>
                  <input
                    class="w-10 bg-transparent text-center underline underline-offset-2 decoration-1 outline-none"
                    id="input"
                    max="999"
                    min="1"
                    name="SET"
                    type="number"
                    value="10"
                  />
                </form>
              </div>
              <div
                class="mx-3 text-base md:text-xl"
              >
                <button
                  id="add"
                  name="ADD"
                >
                  âŠ•
                </button>
              </div>
              <div
                class="delete text-lg md:text-2xl"
              >
                <button
                  id="delete"
                  name="DELETE"
                >
                  ðŸ…§
                </button>
              </div>
            </div>
          </div>
          <div
            class="subtotal flex w-full justify-end items-end space-x-2"
          >
            <div
              class="text-sm md:text-base"
            >
              Subtotal: CAD
            </div>
            <div
              class="font-normal text-lg md:text-2xl"
            >
              0.00
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`<CheckoutItem /> shows content in <CheckoutItem /> 1`] = `
<body>
  <div>
    <div
      class="flex justify-center w-full sm:w-2/3 mx-5 text-slate-700 text-sm md:text-base"
      id="checkoutItem"
    >
      <div
        class="flex w-full bg-opacity-30 bg-slate-100 rounded-lg transition-all duration-200 ease-in-out shadow-md hover:shadow-slate-500"
      >
        <div
          class="image flex flex-col relative w-1/3 h-48"
        >
          <div
            class="relative w-full h-full bg-[#F8F8F8] rounded-l-lg"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="test name image"
                class="object-contain rounded-l-lg"
                data-nimg="fill"
                decoding="async"
                src="test"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </span>
          </div>
        </div>
        <div
          class="text flex flex-col justify-evenly mx-4 w-2/3"
        >
          <div
            class="title flex justify-start"
          >
            <div
              class="font-normal text-lg md:text-2xl"
            >
              test name
            </div>
          </div>
          <div
            class="details flex flex-col md:flex-row items-center justify-between"
          >
            <div
              class="unt flex items-end justify-start text-sm md:text-base text-slate-500"
            >
              <div>
                CAD
              </div>
              <div
                class="text-base md:text-xl ml-1"
              >
                0
              </div>
              <div
                class=""
              >
                /ea
              </div>
            </div>
            <div
              class="qty flex items-center"
            >
              <div
                class="text-sm md:text-base"
              >
                Qty:
              </div>
              <div
                class="mx-3 text-base md:text-xl"
              >
                <button
                  id="remove"
                  name="REMOVE"
                >
                  âŠ–
                </button>
              </div>
              <div
                class="flex justify-center text-base md:text-xl"
              >
                <form>
                  <input
                    class="w-10 bg-transparent text-center underline underline-offset-2 decoration-1 outline-none"
                    id="input"
                    max="999"
                    min="1"
                    name="SET"
                    type="number"
                    value="10"
                  />
                </form>
              </div>
              <div
                class="mx-3 text-base md:text-xl"
              >
                <button
                  id="add"
                  name="ADD"
                >
                  âŠ•
                </button>
              </div>
              <div
                class="delete text-lg md:text-2xl"
              >
                <button
                  id="delete"
                  name="DELETE"
                >
                  ðŸ…§
                </button>
              </div>
            </div>
          </div>
          <div
            class="subtotal flex w-full justify-end items-end space-x-2"
          >
            <div
              class="text-sm md:text-base"
            >
              Subtotal: CAD
            </div>
            <div
              class="font-normal text-lg md:text-2xl"
            >
              0.00
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
